File: src/main/java/cpsc224/items/Item.java
package cpsc224.items;

import javax.swing.*;
import java.awt.*;

public abstract class Item {

    protected String name;
    protected int tier;
    protected ImageIcon icon;

    public Item(String name, int tier, ImageIcon icon) {
        this.name = name;
        this.tier = tier;
        this.icon = icon;
    }

    public Item(String name, int tier) {
        this(name, tier, null);
    }

    /**
     * Gets the icon.
     * @return the icon
     */
    public ImageIcon getIcon() { return icon; }


    /**
     * Maps item tier to its multiplier value.
     * @param tier the item tier
     * @return the multiplier
     */
    public static double mapTierToMultiplier(int tier) {
        return switch (tier) {
            case 2 -> 1.5;
            case 3 -> 2.0;
            default -> 1.0;
        };
    }

    @Override
    public String toString() {
        return name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getTier() {
        return tier;
    }

    public void setTier(int tier){
        this.tier = tier;
    }

    public abstract String getToolTipText();
}

File: web/src/game/items/Item.ts
import type { IconRef } from '../utils/icons';

export abstract class Item {
  protected name: string;
  protected tier: number;
  protected icon: IconRef;

  constructor(name: string, tier: number, icon: IconRef = null) {
    this.name = name;
    this.tier = tier;
    this.icon = icon;
  }

  static mapTierToMultiplier(tier: number): number {
    switch (tier) {
      case 2:
        return 1.5;
      case 3:
        return 2.0;
      default:
        return 1.0;
    }
  }

  getIcon(): IconRef {
    return this.icon;
  }

  getName(): string {
    return this.name;
  }

  setName(name: string): void {
    this.name = name;
  }

  getTier(): number {
    return this.tier;
  }

  setTier(tier: number): void {
    this.tier = tier;
  }

  toString(): string {
    return this.name;
  }

  abstract getToolTipText(): string;
}
